lib/dry/configurable/setting.rb

```diff
@@ -60,8 +60,6 @@ def initialize(name, input: Undefined, default: Undefined, **options)
         @input = input.equal?(Undefined) ? default : input
         @default = default
         @options = options
-
-        evaluate if input_defined?
```

Comment:
I did consider to have a new parameter to evaluate conditionally, but IMO this class already has too many parameters.
The evaluation is only needed in one place, so removing this conditional and do the evaluation only when needed seems like a better fit for the requirement.